hypothesis-python/src/hypothesis/extra/pytestplugin.py

```diff
@@ -120,42 +124,36 @@ def pytest_runtest_makereport(item, call):
             'Hypothesis',
             '\n'.join(item.hypothesis_report_information)
         ))
+    if hasattr(item, 'hypothesis_statistics') and report.when == 'teardown':
+        # Running on pytest <= 3.5 where user_properties doesn't exist, fall
+        # back on the global gathered_statistics (which breaks under xdist)
+        if hasattr(report, 'user_properties'):  # pragma: no branch
+            report.user_properties += (
+                ('hypothesis-stats', item.hypothesis_statistics),
+            )
 
 
 def pytest_terminal_summary(terminalreporter):
     if not terminalreporter.config.getoption(PRINT_STATISTICS_OPTION):
         return
     terminalreporter.section('Hypothesis Statistics')
-    for name, statistics in gathered_statistics.items():
-        terminalreporter.write_line(name + ':')
-        terminalreporter.write_line('')
 
-        if not statistics.has_runs:
-            terminalreporter.write_line('  - Test was never run')
-            continue
+    if LooseVersion(pytest.__version__) < '3.5':  # pragma: no cover
+        if terminalreporter.config.getoption('-n'):
```

Comment:
This feels like a very fragile way to check if a plugin is enabled. Is there really not a better one?

Reply:
Nope :sob: 

We can robustly check if the `xdist` plugin is *installed*, and we can robustly check if the current process one of the workers (and if so which one), but there's no way I can find to distinguish between being the master process and `xdist` not being enabled.

(I did add a check for the `--numprocesses` form, even though I've never seen it in the wild)